genesyscloud/architect_schedules/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,262 @@
+	if getErr != nil {
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of schedule error: %s", getErr), proxyResponse)
+	}
+
+	for _, schedule := range *schedules {
+		resources[*schedule.Id] = &resourceExporter.ResourceMeta{Name: *schedule.Name}
+	}
+
+	return resources, nil
+}
+
+// createArchitectSchedulegs is used by the architect_schedulegs resource to create Genesys cloud architect schedules
+func createArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	divisionID := d.Get("division_id").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", start, err)
+	}
+
+	schedEnd, err := time.Parse("2006-01-02T15:04:05.000000", end)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", end, err)
+	}
+
+	schedule := platformclientv2.Schedule{
+		Name:  &name,
+		Start: &schedStart,
+		End:   &schedEnd,
+		Rrule: &rrule,
+	}
+
+	// Optional attributes
+	if description != "" {
+		schedule.Description = &description
+	}
+
+	if divisionID != "" {
+		schedule.Division = &platformclientv2.Writabledivision{Id: &divisionID}
+	}
+
+	log.Printf("Creating schedule %s", *schedule.Name)
+	scheduleResponse, proxyResponse, err := proxy.createArchitectSchedules(ctx, &schedule)
+	if err != nil {
+		msg := ""
+		if strings.Contains(fmt.Sprintf("%v", err), "routing:schedule:add") {
+			msg = "\nYou must have all divisions and future divisions selected in your OAuth client role"
+		}
+
+		return util.BuildAPIDiagnosticError("genesyscloud_archiect_schedules", fmt.Sprintf("Failed to create schedule %s | Error: %s MSG: %s", *scheduleResponse.Name, err, msg), proxyResponse)
+	}
+
+	d.SetId(*scheduleResponse.Id)
+
+	log.Printf("Created schedule %s %s", name, *scheduleResponse.Id)
+	return readArchitectSchedules(ctx, d, meta)
+}
+
+func readArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	log.Printf("Reading schedule %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		scheduleResponse, proxyResponse, err := proxy.getArchitectSchedulesById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(proxyResponse) {
+				return retry.RetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read schedule %s: %s", d.Id(), err))
+		}
+
+		Start := new(string)
```

Comment:
As a go standard we do not name local variables with a Capital.  In go, a struct attribute with a Capital letter is interpreted to be public.  If it is lower case it is private.  In the context of a local variable you don't have a public/private distinction but it is still the norm to keep it lower case.

Also, you can just do Start:="" instead of Start :=new(String)